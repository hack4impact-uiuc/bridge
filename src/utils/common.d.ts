declare module "@hack4impact-uiuc/bridge/common" {
  import { Props, ReactType } from "react";
  import {
    ColorProps,
    SpaceProps,
    LayoutProps,
    TypographyProps as StyledTypographyProps,
    BordersProps as StyledBordersProps,
    BoxShadowProps,
  } from "styled-system";

  export interface BaseProps extends Props<any> {
    as?: ReactType;
    className?: string;
  }
  export interface CommonProps
    extends BaseProps,
      ColorProps,
      SpaceProps,
      LayoutProps {}

  export interface TypographyProps extends BaseProps, StyledTypographyProps {}

  export interface BorderProps extends StyledBordersProps, BoxShadowProps {}
}
