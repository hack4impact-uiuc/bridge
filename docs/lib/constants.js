export const COMPONENTS_URL = '/components';
export const DESIGN_URL = '/design';
