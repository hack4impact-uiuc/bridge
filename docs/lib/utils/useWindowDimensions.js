import { useState, useEffect } from 'react';

function getWindowDimensions() {
  const { innerWidth: width, innerHeight: height } = window;
  return {
    width,
    height,
  };
}

export default function useWindowDimensions() {
  const [windowDimensions, setWindowDimensions] = useState(getWindowDimensions());

  useEffect(() => {
    function handleResize() {
      const { innerWidth: width, innerHeight: height } = window;
      setWindowDimensions({ width, height });
    }

    window.addEventListener('resize', handleResize);
    return () => window.removeEventListener('resize', handleResize);
  }, []);

  return windowDimensions;
}
