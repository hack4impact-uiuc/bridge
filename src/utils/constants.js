import * as styledSystem from 'styled-system';
import systemPropTypes from '@styled-system/prop-types';
import PropTypes from 'prop-types';


const { compose, system } = styledSystem;

// Text Decoration Prop
export const textDecoration = system({
  prop: 'textDecoration',
  cssProperty: 'text-decoration',
});
textDecoration.propTypes = PropTypes.string;


// common style props to be passed into components
export const COMMON = compose(styledSystem.space, styledSystem.layout);
COMMON.propTypes = {
  ...systemPropTypes.space,
  ...systemPropTypes.layout,
};

// for easier access
export const TYPOGRAPHY = styledSystem.typography;
export const LAYOUT = styledSystem.layout;
export const POSITION = styledSystem.position;
export const FLEX = styledSystem.flexbox;
export const GRID = styledSystem.grid;
export const COLOR = styledSystem.color;
export const BORDER = styledSystem.border;

TYPOGRAPHY.propTypes = systemPropTypes.typography;
LAYOUT.propTypes = systemPropTypes.layout;
POSITION.propTypes = systemPropTypes.position;
FLEX.propTypes = systemPropTypes.flexbox;
GRID.propTypes = systemPropTypes.grid;
COLOR.propTypes = systemPropTypes.color;
BORDER.propTypes = systemPropTypes.border;
