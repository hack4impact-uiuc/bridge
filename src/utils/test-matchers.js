/* eslint-disable react/forbid-foreign-prop-types */
// taken from https://github.com/primer/components/blob/master/src/utils/test-matchers.js
import 'jest-styled-components';

const stringify = (d) => JSON.stringify(d, null, '  ');
/**
 * These are props that styled-system aliases for backwards compatibility.
 * For some reason, they don't show up in our toImplementSystemProps() matcher,
 * so we skip over them.
 */
const ALIAS_PROP_TYPES = ['w', 'align', 'justify', 'wrap'];

// eslint-disable-next-line no-undef
expect.extend({
  toImplementSystemProps(Component, propNames) {
    const propKeys = new Set(Object.keys(Component.propTypes));
    const expectedPropKeys = Object.keys(propNames.propTypes);
    const missing = expectedPropKeys.filter((key) => !propKeys.has(key))
      .filter((key) => !ALIAS_PROP_TYPES.includes(key));
    return {
      pass: missing.length === 0,
      message: () => `Missing prop${missing.length === 1 ? '' : 's'}: ${stringify(missing)}`,
    };
  },
});
