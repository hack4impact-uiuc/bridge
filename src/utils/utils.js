import themeGet from '@styled-system/theme-get';
import * as styledSystem from 'styled-system';
import theme from '../theme';

const { get: getKey } = styledSystem;

export const get = (key) => themeGet(key, getKey(theme, key));

export const getOrDefault = (val, def) => (val !== undefined && val !== null ? val : def);

export const lodashGet = (object, path, defaultVal) => {
  const PATH = Array.isArray(path)
    ? path
    : path.split('.').filter((i) => i.length);
  if (!PATH.length) {
    return object === undefined ? defaultVal : object;
  }
  if (object === null || object === undefined || typeof (object[PATH[0]]) === 'undefined') {
    return defaultVal;
  }
  return lodashGet(object[PATH.shift()], PATH, defaultVal);
};
