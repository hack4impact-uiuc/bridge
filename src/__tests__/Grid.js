// taken from https://github.com/primer/components/blob/master/src/__tests__/Grid.js#L7
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { axe, toHaveNoViolations } from 'jest-axe';

import { Grid } from '..';
import { GRID } from '../utils/constants';
import { renderJSON } from '../utils/testing';
import 'babel-polyfill'; // axe violations required babel-polyfill

expect.extend(toHaveNoViolations);

describe('Grid', () => {
  it('should have no axe violations', async () => {
    const { container } = render(<Grid />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    cleanup();
  });

  it('implements system props', () => {
    expect(Grid).toImplementSystemProps(GRID);
  });

  it('gets display: grid by default', () => {
    expect(renderJSON(<Grid />)).toHaveStyleRule('display', 'grid');
  });

  it('respects gridGap', () => {
    expect(renderJSON(<Grid gridGap={1} />)).toMatchSnapshot();
  });

  it('respects gridColumnGap', () => {
    expect(renderJSON(<Grid gridColumnGap={2} />)).toMatchSnapshot();
  });

  it('respects gridRowGap', () => {
    expect(renderJSON(<Grid gridRowGap={2} />)).toMatchSnapshot();
  });

  it('respects gridColumn', () => {
    expect(renderJSON(<Grid gridColumn="1 / 2" />)).toMatchSnapshot();
  });

  it('respects gridRow', () => {
    expect(renderJSON(<Grid gridRow="1 / 2" />)).toMatchSnapshot();
  });

  it('respects gridArea', () => {
    expect(renderJSON(<Grid gridArea="sidebar" />)).toMatchSnapshot();
  });

  it('respects gridAutoFlow', () => {
    expect(renderJSON(<Grid gridAutoFlow="row" />)).toMatchSnapshot();
  });

  it('respects gridAutoRows', () => {
    expect(renderJSON(<Grid gridAutoRows="1fr" />)).toMatchSnapshot();
  });

  it('respects gridAutoColumns', () => {
    expect(renderJSON(<Grid gridAutoColumns="1fr" />)).toMatchSnapshot();
  });

  it('respects gridTemplateColumns', () => {
    expect(renderJSON(<Grid gridTemplateColumns="200px 1fr" />)).toMatchSnapshot();
  });

  it('respects gridTemplateRows', () => {
    expect(renderJSON(<Grid gridTemplateRows=" 200px 1fr" />)).toMatchSnapshot();
  });

  it('respects gridTemplateAreas', () => {
    expect(renderJSON(<Grid gridTemplateAreas="sidebar  main" />)).toMatchSnapshot();
  });

  it('respects responsive display', () => {
    expect(renderJSON(<Grid display={['grid', 'inline-grid']} />)).toMatchSnapshot();
  });

  it('respects the "as" prop', () => {
    expect(renderJSON(<Grid as="span" />).type).toEqual('span');
  });

  it('renders a div by default', () => {
    expect(renderJSON(<Grid />).type).toEqual('div');
  });
});
