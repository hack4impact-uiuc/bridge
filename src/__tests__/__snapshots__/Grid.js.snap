// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid respects gridArea 1`] = `
.c0 {
  display: grid;
  grid-area: sidebar;
}

<div
  className="Box-ik9o9h-0 c0"
  display="grid"
/>
`;

exports[`Grid respects gridAutoColumns 1`] = `
.c0 {
  display: grid;
  grid-auto-columns: 1fr;
}

<div
  className="Box-ik9o9h-0 c0"
  display="grid"
/>
`;

exports[`Grid respects gridAutoFlow 1`] = `
.c0 {
  display: grid;
  grid-auto-flow: row;
}

<div
  className="Box-ik9o9h-0 c0"
  display="grid"
/>
`;

exports[`Grid respects gridAutoRows 1`] = `
.c0 {
  display: grid;
  grid-auto-rows: 1fr;
}

<div
  className="Box-ik9o9h-0 c0"
  display="grid"
/>
`;

exports[`Grid respects gridColumn 1`] = `
.c0 {
  display: grid;
  grid-column: 1 / 2;
}

<div
  className="Box-ik9o9h-0 c0"
  display="grid"
/>
`;

exports[`Grid respects gridColumnGap 1`] = `
.c0 {
  display: grid;
  grid-column-gap: 8px;
}

<div
  className="Box-ik9o9h-0 c0"
  display="grid"
/>
`;

exports[`Grid respects gridGap 1`] = `
.c0 {
  display: grid;
  grid-gap: 4px;
}

<div
  className="Box-ik9o9h-0 c0"
  display="grid"
/>
`;

exports[`Grid respects gridRow 1`] = `
.c0 {
  display: grid;
  grid-row: 1 / 2;
}

<div
  className="Box-ik9o9h-0 c0"
  display="grid"
/>
`;

exports[`Grid respects gridRowGap 1`] = `
.c0 {
  display: grid;
  grid-row-gap: 8px;
}

<div
  className="Box-ik9o9h-0 c0"
  display="grid"
/>
`;

exports[`Grid respects gridTemplateAreas 1`] = `
.c0 {
  display: grid;
  grid-template-areas: sidebar main;
}

<div
  className="Box-ik9o9h-0 c0"
  display="grid"
/>
`;

exports[`Grid respects gridTemplateColumns 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 200px 1fr;
}

<div
  className="Box-ik9o9h-0 c0"
  display="grid"
/>
`;

exports[`Grid respects gridTemplateRows 1`] = `
.c0 {
  display: grid;
  grid-template-rows: 200px 1fr;
}

<div
  className="Box-ik9o9h-0 c0"
  display="grid"
/>
`;

exports[`Grid respects responsive display 1`] = `
.c0 {
  display: grid;
}

@media screen and (min-width:544px) {
  .c0 {
    display: inline-grid;
  }
}

<div
  className="Box-ik9o9h-0 c0"
  display={
    Array [
      "grid",
      "inline-grid",
    ]
  }
/>
`;
